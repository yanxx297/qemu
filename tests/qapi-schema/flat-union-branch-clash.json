# FIXME: we should check for no duplicate keys between branches and base
{ 'enum': 'TestEnum',
  'data': [ 'value1', 'value2' ] }
{ 'type': 'Base',
  'data': { 'enum1': 'TestEnum', 'name': 'str' } }
{ 'type': 'Branch1',
  'data': { 'name': 'str' } }
{ 'type': 'Branch2',
  'data': { 'value': 'int' } }
{ 'union': 'TestUnion',
  'base': 'Base',
  'discriminator': 'enum1',
  'data': { 'value1': 'Branch1',
            'value2': 'Branch2' } }
