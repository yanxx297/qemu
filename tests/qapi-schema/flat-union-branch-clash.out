[OrderedDict([('enum', 'TestEnum'), ('data', ['value1', 'value2'])]),
 OrderedDict([('type', 'Base'), ('data', OrderedDict([('enum1', 'TestEnum'), ('name', 'str')]))]),
 OrderedDict([('type', 'Branch1'), ('data', OrderedDict([('name', 'str')]))]),
 OrderedDict([('type', 'Branch2'), ('data', OrderedDict([('value', 'int')]))]),
 OrderedDict([('union', 'TestUnion'), ('base', 'Base'), ('discriminator', 'enum1'), ('data', OrderedDict([('value1', 'Branch1'), ('value2', 'Branch2')]))])]
[{'enum_name': 'TestEnum', 'enum_values': ['value1', 'value2']}]
[OrderedDict([('type', 'Base'), ('data', OrderedDict([('enum1', 'TestEnum'), ('name', 'str')]))]),
 OrderedDict([('type', 'Branch1'), ('data', OrderedDict([('name', 'str')]))]),
 OrderedDict([('type', 'Branch2'), ('data', OrderedDict([('value', 'int')]))])]
